cmd/notary/integration_test.go

```diff
@@ -183,6 +183,131 @@ func TestInitWithRootKey(t *testing.T) {
+MIIBWDCB/6ADAgECAhBKKoVsRNJdGsGh6tPWnE4rMAoGCCqGSM49BAMCMBMxETAP
+BgNVBAMTCGRvY2tlci8qMB4XDTE3MDQyODIwMTczMFoXDTI3MDQyNjIwMTczMFow
+EzERMA8GA1UEAxMIZG9ja2VyLyowWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQQ
+6RhA8sX/kWedbPPFzNqOMI+AnWOQV+u0+FQfeNO+k/Uf0LBnKhHEPSwSBuuwLPon
+w+nR0YTdv3lFaM7x9nOUozUwMzAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYI
+KwYBBQUHAwMwDAYDVR0TAQH/BAIwADAKBggqhkjOPQQDAgNIADBFAiA+eHPInhLJ
+HgP8nha+UqdYgq8ZCOlhdGTJhSdHd4sCuQIhAPXqQeWhDLA3/Pf8B7G3ZwWpPbZ8
+adLwkjqoeEKMaAXf
+-----END CERTIFICATE-----`
+
+	nonMatchingKeyStr := `-----BEGIN EC PRIVATE KEY-----
+Proc-Type: 4,ENCRYPTED
+DEK-Info: AES-256-CBC,fd6e6735232efbc1a851549d12b8203d
+role: root
+
+Z6u+cAZOEmeoieyQHt6Lp8ZmLWPiyGXT0wTkfYMnGxZ+EX+6sBeu9CWgx+3kOCWQ
+qXuLmBjJ4ZwL/lZejeLLefF7jILA0oDLJtNH1L0oP7H/i7DUtNv+7Jvnci986Rx0
+i85wnaTwOgWv8n6q3tavmnIA/v2QqsTpmI+bhwrPNKQ=
+-----END EC PRIVATE KEY-----`
+
+	//set up notary server
+	server := setupServer()
+	defer server.Close()
+
+	//set up temp dir
+	tempDir := tempDirWithConfig(t, `{
+		"trust_pinning" : {
+			"disable_tufu" : false
+		}
+	}`)
+	defer os.RemoveAll(tempDir)
+
+	//test tempfile writable
+	tempFile, err := ioutil.TempFile("", "targetfile")
+	require.NoError(t, err)
+	tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	gun := "docker/repoName"
+	privKeyFilename := filepath.Join(tempDir, "priv.key")
+	certFilename := filepath.Join(tempDir, "cert.pem")
+	nonMatchingKeyFilename := filepath.Join(tempDir, "nmkey.key")
+
+	//write key and cert to file
+	err = ioutil.WriteFile(privKeyFilename, []byte(privStr), 0644)
+	require.NoError(t, err)
+	err = ioutil.WriteFile(certFilename, []byte(certStr), 0644)
+	require.NoError(t, err)
+	err = ioutil.WriteFile(nonMatchingKeyFilename, []byte(nonMatchingKeyStr), 0644)
+	require.NoError(t, err)
+
+	//test init repo without --rootkey and --rootcert
+	output, err := runCommand(t, tempDir,
+		"-s", server.URL,
+		"init", gun+"1",
+		"--rootkey", privKeyFilename,
+		"--rootcert", certFilename)
+	require.NoError(t, err)
+	require.Contains(t, output, "Root key found")
+	// === no rootkey specified: look up in keystore ===
+	// this requires the previous test to inject private key in keystore
+	output, err = runCommand(t, tempDir,
+		"-s", server.URL,
+		"init", gun+"2",
+		"--rootcert", certFilename)
+	require.NoError(t, err)
+	require.Contains(t, output, "Root key found")
+
+	//add a file to repo
+	output, err = runCommand(t, tempDir,
+		"-s", server.URL,
+		"add", gun+"2",
+		"v1",
+		certFilename)
+	require.NoError(t, err)
+	require.Contains(t, output, "staged for next publish")
+
+	//publish repo
+	output, err = runCommand(t, tempDir,
+		"-s", server.URL,
+		"publish", gun+"2")
+	require.NoError(t, err)
+	require.Contains(t, output, "Successfully published changes")
+
+	// === test init with no argument to --rootcert ===
+	_, err = runCommand(t, tempDir,
+		"-s", server.URL,
+		"init", gun+"3",
+		"--rootkey", privKeyFilename,
+		"--rootcert")
+	require.Error(t, err)
```

Comment:
Can we add some checks on the output here to make sure the messages returned to the user are useful?